Src/DependencyCollector/Shared/Implementation/SqlClientDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,539 @@
+                Id = operationId.ToString("N"),
+                Name = dependencyName,
+                Type = RemoteDependencyConstants.SQL,
+                Target = target,
+                Data = command.CommandText,
+                Duration = duration,
+                Timestamp = DateTimeOffset.UtcNow - duration,
+                Success = true
+            };
+
+            InitializeTelemetry(telemetry, operationId);
+
+            return telemetry;
+        }
+
+        private void OnErrorConnectionEvent(
+            Guid operationId,
+            string operation,
+            Guid connectionId,
+            SqlConnection connection,
+            Exception exception,
+            long endTimestamp)
+        {
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+
+            var telemetry = new DependencyTelemetry()
+            {
+                Id = operationId.ToString("N"),
+                Name = string.Join(" | ", connection.DataSource, connection.Database, operation),
+                Type = RemoteDependencyConstants.SQL,
+                Target = string.Join(" | ", connection.DataSource, connection.Database),
+                Data = operation,
+                Duration = duration,
+                Timestamp = DateTimeOffset.UtcNow - duration,
+                Success = true
+            };
+
+            InitializeTelemetry(telemetry, operationId);
+            ConfigureExceptionTelemetry(telemetry, exception);
+
+            this.client.Track(telemetry);
+        }
+
+        private void OnAfterTransactionEvent(
+            Guid operationId,
+            string operation,
+            IsolationLevel isolationLevel,
+            SqlConnection connection,
+            long endTimestamp)
+        {
+            var telemetry = this.CreateTransactionTelemetry(operationId, operation, isolationLevel, connection, endTimestamp);
+
+            this.client.Track(telemetry);
+        }
+
+        private void OnErrorTransactionEvent(
+            Guid operationId,
+            string operation,
+            IsolationLevel isolationLevel,
+            SqlConnection connection,
+            Exception exception,
+            long endTimestamp)
+        {
+            var telemetry = this.CreateTransactionTelemetry(operationId, operation, isolationLevel, connection, endTimestamp);
+
+            ConfigureExceptionTelemetry(telemetry, exception);
+
+            this.client.Track(telemetry);
+        }
+
+        private DependencyTelemetry CreateTransactionTelemetry(
+            Guid operationId, string operation, IsolationLevel isolationLevel, SqlConnection connection, long endTimestamp)
+        {
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+
+            var telemetry = new DependencyTelemetry()
+            {
+                Id = operationId.ToString("N"),
+                Name = string.Join(" | ", connection.DataSource, connection.Database, operation, isolationLevel),
```

Comment:
what would be the name of the operation here?

Reply:
"Commit" or "Rollback"